### ISSUE TITLE
datasets

### ISSUE THREAD

**Issue by gnftau** [meituan/YOLOv6#84]
My training set has over 3,000 images, but only 275 valid images were displayed during yolov6 training. No similar situation occurs in YoloV5.

**Comment by Zephyr69** [comment]
Similar situation here, but with 0 valid images. I think the dataset format is exactly the same as yolov5's, with minor adjustment in yaml.

All exceptions are "ignoring corrupt image: 'NoneType' object has no attribute 'seek'". I couldn't figure it out.

**Comment by Zephyr69** [comment]
Ok, after playing around a bit and finally started training, I found something that may help you.

1. This split will cause problems if you have '.' in the filename. Using pathlib will resolve this problem.
https://github.com/meituan/YOLOv6/blob/1d1335c62a939f2e5d639bdf2afec65c18beb59f/yolov6/data/datasets.py#L253

2. This returns None. I don't know how to address this but setting it to False and adding image format manually.
https://github.com/meituan/YOLOv6/blob/1d1335c62a939f2e5d639bdf2afec65c18beb59f/yolov6/data/datasets.py#L416

3. When an exception is raised, this may not actually ignore the corrupted label but break the dataloader entirely.
https://github.com/meituan/YOLOv6/blob/1d1335c62a939f2e5d639bdf2afec65c18beb59f/yolov6/data/datasets.py#L474

**Comment by gnftau** [comment]
After my inquiry, I finally found the problem. In the process of verifying the image, there is a PIL object's verify () function, which verifies the integrity of the image, annotating this function to load the entire data set. Incidentally, you also need to delete the JSON files generated by the dataset, or the program won't reload the methods that validate the data.

YOLOv6/yolov6/data/datasets.py     
def check_image(im_file):
       ……
       im.verify()     #kill

**Comment by Zephyr69** [comment]
> After my inquiry, I finally found the problem. In the process of verifying the image, there is a PIL object's verify () function, which verifies the integrity of the image, annotating this function to load the entire data set. Incidentally, you also need to delete the JSON files generated by the dataset, or the program won't reload the methods that validate the data.
> 
> YOLOv6/yolov6/data/datasets.py def check_image(im_file): …… im.verify() #kill

Commenting that line also worked!
